### DISCUSSION TITLE
Type mismatches for indentation levels in tree-sitter-python scanner.c

### DISCUSSION THREAD

**Discussion by osa1** [tree-sitter/tree-sitter#3526] [Q&A]
tree-sitter-python's issue template won't let me create this as an issue, so creating a discussion in this repo instead.

- The array for indentation levels uses 2 bytes per element: https://github.com/tree-sitter/tree-sitter-python/blob/0dee05ef958ba2eae88d1e65f24b33cad70d4367/src/scanner.c#L85-L89

- `indent_length` local that holds the indentation of the current line uses a 4 byte integer: https://github.com/tree-sitter/tree-sitter-python/blob/0dee05ef958ba2eae88d1e65f24b33cad70d4367/src/scanner.c#L214

- Lexer state serialization and deserialization functions treat the indentation level array elements as one byte: 
  - Serialization: https://github.com/tree-sitter/tree-sitter-python/blob/0dee05ef958ba2eae88d1e65f24b33cad70d4367/src/scanner.c#L383
  - Deserialization: https://github.com/tree-sitter/tree-sitter-python/blob/0dee05ef958ba2eae88d1e65f24b33cad70d4367/src/scanner.c#L410

Shouldn't all of these be the same type?

**Comment by amaanq** [comment]
Yeah, that's a good point, especially for the (de)serialization. Typically we'd write 1 byte because the max # of bytes you can write to the buffer is 1kb, but I don't think we'd realistically hit that limit, even with 2-byte indents (giving us a max of 512 indents being stored). I've submitted a PR [here](https://github.com/tree-sitter/tree-sitter-python/pull/294), thanks for the callout
